src/java/org/apache/cassandra/db/streaming/CassandraStreamReader.java

```diff
@@ -136,10 +143,10 @@ public SSTableMultiWriter read(DataInputPlus inputPlus) throws IOException
             logger.warn("[Stream {}] Error while reading partition {} from stream on ks='{}' and table='{}'.",
                         session.planId(), partitionKey, cfs.keyspace.getName(), cfs.getTableName(), e);
             if (writer != null)
-            {
-                writer.abort(e);
-            }
-            throw Throwables.propagate(e);
+                e = writer.abort(e);
+            Throwables.throwIfUnchecked(e);
+            Throwables.throwIfInstanceOf(e, Exception.class);
+            throw new RuntimeException(e); // not possible; just here to compile
```

Comment:
At this point, what if we just do the following:

1.) Leave all the method signatures alone in the `IStreamReader` hierarchy.

2.) Change the exception we throw for dropped tables from `IOException` to something a bit more appropriate, like maybe `IllegalStateException`.

3.) Change the bits above to just:

```
Throwables.throwIfUnchecked(e);
if (e instanceof IOException) throw (IOException) e;
throw new RuntimeException(e);
```

My guess is we wanted to avoid wrapping `IOException` in a `RuntimeException`, but as things stand, the `throws Exception` doesn't really help explain the API (which is probably the only point of checked exceptions in the first place).

Reply:
If we did that, we could also revert the methods in the `IncomingStream` hierarchy to keep `throws IOException`.